src/Surfnet/Stepup/Identity/Event/SecondFactorEvent.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/**
+ * Copyright 2014 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\Stepup\Identity\Event;
+
+use Surfnet\Stepup\Identity\Value\IdentityId;
+use Surfnet\Stepup\Identity\Value\Institution;
+use Surfnet\Stepup\Identity\Value\SecondFactorId;
+use Surfnet\StepupBundle\Value\SecondFactorType;
+
+abstract class SecondFactorEvent extends IdentityEvent
```

Comment:
Not sure I like this...(the naming that is). I get the extension, but the simple "{x} is a {y} rule" doesn't really work here with the semantics of the domain, especially when seeing which events are the concrete implementations. `EmailVerifiedEvent`, `CompliedWithRevocationEvent` are specific events that register actions done by the Identity (The identity verified the email, the identity complied with the revocation). How about renaming it to something like `SecondFactorAwareEvent`, or even simpler, `AuditableEvent` - the latter could enforce the presence of required fields for the auditlog... thoughts?